Src/ILGPU.Tests/ExchangeBufferOperations.cs

```diff
@@ -0,0 +1,204 @@
+            {
+                for (int j = 0; j < Length; j++)
+                {
+                    exchangeBuffer[new Index2(i, j)] = constant;
+                }
+            }
+            exchangeBuffer.CopyToAccelerator();
+
+            var expected = new long[Length][];
+            for (int i = 0; i < Length; i++)
+            {
+                expected[i] = new long[Length];
+                for (int j = 0; j < Length; j++)
+                {
+                    expected[i][j] = constant - 5;
+                }
+            }
+
+            Accelerator.Synchronize();
+            Execute(exchangeBuffer.Extent, exchangeBuffer.View);
+            exchangeBuffer.CopyFromAccelerator();
+            Accelerator.Synchronize();
+
+            var data = exchangeBuffer.CPUView;
+            Assert.Equal(data.Extent.X, expected.Length);
+            for (int i = 0; i < Length; i++)
+            {
+                Assert.Equal(data.Extent.Y, expected[i].Length);
+                for (int j = 0; j < Length; j++)
+                {
+                    Assert.Equal(expected[i][j], data[new Index2(i, j)]);
+                }
+            }
+        }
+
+        internal static void GetAsArray2DKernel(Index2 index, ArrayView<long, Index2> data, long c)
+        {
+            data[index] = c;
+        }
+
+        [Theory]
+        [InlineData(10)]
+        [InlineData(int.MaxValue)]
+        [InlineData(int.MinValue)]
+        [KernelMethod(nameof(GetAsArray2DKernel))]
+        public void GetAsArray2D(long constant)
+        {
+            using var exchangeBuffer = Accelerator.AllocateExchangeBuffer<long>(new Index2(Length, Length));
+            Execute(exchangeBuffer.Extent, exchangeBuffer.View, constant);
+
+            var expected = new long[Length][];
+            for (int i = 0; i < Length; i++)
+            {
+                expected[i] = new long[Length];
+                for (int j = 0; j < Length; j++)
+                {
+                    expected[i][j] = constant;
+                }
+            }
+
+            var data = exchangeBuffer.GetAs2DArray(Accelerator.DefaultStream);
+            Assert.Equal(data.Length, expected.Length);
+            for (int i = 0; i < Length; i++)
+            {
+                Assert.Equal(data[i].Length, expected[i].Length);
+                for (int j = 0; j < Length; j++)
+                {
+                    Assert.Equal(data[i][j], expected[i][j]);
+                }
+            }
+        }
+
+        internal static void Copy3DKernel(Index3 index, ArrayView<long, Index3> data)
+        {
+            data[index] -= 5;
+        }
+
+        [Theory]
+        [InlineData(10)]
+        [InlineData(int.MaxValue)]
+        [InlineData(int.MinValue)]
+        [KernelMethod(nameof(Copy3DKernel))]
+        public void Copy3D(long constant)
+        {
+            using var exchangeBuffer = Accelerator.AllocateExchangeBuffer<long>(new Index3(Length, Length, Length));
```

Comment:
Try to keep the maximum line of 90 characters. This is not a strict limitations for the tests classes. However, it might look nicer and easier to read 🤓

Reply:
I'll try to come up with simpler tests